### PR TITLE
[ADD] new module to create supplier claim in customer claim.

### PR DESCRIPTION
**Pull request by yaniaular**

### CODE REVIEW

**Comment by yaniaular** [comment]
We are working on this PR to make it perfect to merge with OCA/rma.
The modules presented here are part of a collection of modules for the operations of RMA improved by Vauxoo. Each module will be presented in separate PRs because the PR #26 was too heavy for review.. Some dependencies needed.

File: crm_rma_claim_make_claim/README.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+CRM RMA Claim make Claim
```
Comment (pedrobaeza) [inline crm_rma_claim_make_claim/README.rst]: Use OCA's template

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+
+from openerp import fields, models, api
+from openerp.exceptions import except_orm
+from openerp.tools.translate import _
+
+
+class claim_line(models.Model):
```
Comment (pedrobaeza) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: CamelCase

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                              " - For a Custtomer Claim means"
+                              " the RMA-V generated to"
+                              " fullfill the current RMA-C.")
+
+    @api.multi
+    def button_create_rma_vendor(self):
+        """
+        Create RMA Vendor.
+        """
+        # TODO know serial/lot number with
```
Comment (pedrobaeza) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: Put this TODO in the "Known issues / Roadmap" section

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        # depends of supplier module
+
+        claim_line_obj = self.env['claim.line']
+        # inv_obj = self.env['account.invoice']
+        # invline_obj = self.env['account.invoice.line']
+        good_lines = []
+
+        for claim_brw in self:
+
+            if claim_brw.claim_ids:
```
Comment (pedrobaeza) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: You can make `if self.mapped('claim_ids'):`

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                    claim_lines_exists = claim_line_obj.search(
+                        [('claim_line_id', '=', claim_line_id)],)
+                    claim_lines_exists = [cla.id for cla in claim_lines_exists]
+
+                    # If claim_line_id does not have child
+                    if not claim_lines_exists:
+                        # The claim line of supplier claim is created
+                        claim_line_id = claim_line_obj.browse(claim_line_id)
+
+                        # invl = \
```
Comment (pedrobaeza) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: Remove commented code

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                         'claim_rel',
+                         'claim_parent',
+                         'claim_child',
+                         string="Related Claims",
+                         help=" - For a Vendor Claim means"
+                              " the RMA-C that generate the"
+                              " current RMA-V.\n"
+                              " - For a Custtomer Claim means"
+                              " the RMA-V generated to"
+                              " fullfill the current RMA-C.")
```
Comment (LeartS) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: The entire field definition has a really wasteful line breaking!

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+                    _('The Supplier Claim Was Not Created.'))
+            good_lines = list(set(good_lines))
+            mod_obj = self.env['ir.model.data']
+            act_obj = self.env['ir.actions.act_window']
+
+            result = mod_obj.get_object_reference('crm_claim_rma',
+                                                  'act_crm_case_claim_lines')
+            act_id = result and result[1] or False
+            act_rec = act_obj.browse(act_id)
+            result = act_rec.read()
```
Comment (LeartS) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: these 7 lines can be replaced by `self.env.ref('crm_claim_rma.act_crm_case_claim_lines')` [`.ref` doc](https://www.odoo.com/documentation/8.0/reference/orm.html#common-orm-methods)

File: crm_rma_claim_make_claim/model/crm_claim_rma.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                                # 'supplier_invoice_id': inv_obj,
+                                # 'invoice_line_id': invl[0].id,
+                            })
+
+                        # is added for display in the view
+                        good_lines.append(claim_line_new.id)
+                    else:
+                        # Anyway. If claim line have child,
+                        # is added for display in the view
+                        good_lines += claim_lines_exists
```
Comment (LeartS) [inline crm_rma_claim_make_claim/model/crm_claim_rma.py]: is it me or is this entire for loop really strange? It groups claim_lines by partner_id and then it completely ignores the grouping and just works on all claim lines normally..

**Comment by gurneyalex** [comment]
CI red -> needs fixing.

**Comment by osvalr** [comment]
@yaniaular test have been changed

**Comment by nhomar** [comment]
@gurneyalex @pedrobaeza You feedback is needed here.

for me +1

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**